release_notes/ocp-4-2-release-notes.adoc

```diff
@@ -0,0 +1,890 @@
+xref:../nodes/nodes/nodes-nodes-working.adoc[Working with nodes] for more information.
+
+[id="ocp-4-2-infrastructure"]
+=== Infrastructure
+
+[id="ocp-4-2-rhhi-next"]
+==== RHHI.Next
+
+RHHI.Next is a unified platform for container and VM workloads under common
+management control. Storage, compute, and networking is tightly integrated into
+{product-title} for a seamless customer experience. This includes:
+
+* {product-title}
+* OpenShift Container Storage
+* Container-native Virtualization
+* RHEL CoreOS
+* Industry Standard Server Hardware
+* Network Switches (optional)
+
+Benefits:
+
+* Get the infrastructure for your on-premise application development up and running quickly.
+* Hyperconverged infrastructure makes it easier to order, install, and manage.
+* Run virtualized workloads side-by-side with newer container workloads.
+* Consistent operational management with public cloud using {product-title}.
+
+[id="ocp-4-2-networking"]
+=== Networking
+
+==== Installer-provisioned OpenShift on OpenStack
+
+{product-title} 4.2 introduces full-stack automation support for deploying
+{product-title} clusters on Red Hat OpenStack.
+
+The installer uses the OpenStack APIs in conjunction with the Kubernetes
+OpenStack Cloud Provider to create all the required OpenStack resources, such as
+virtual machines (VMs), networks, security groups, and object storage needed to
+deploy {product-title} and properly configure the cluster to run on the Red Hat
+OpenStack Platform.
+
+{product-title} 4.2 can be deployed on Red Hat OpenStack version 13.
+
+Kuryr exists to improve the network performance of pods when running workloads
+on Red Hat OpenStack. Kuryr is an SDN solution combining a Kubernetes CNI plug-in
+(kuryr-kubernetes) and OpenStack Neutron to provide interconnectivity between
+Kubernetes pods and OpenStack virtual instances.
+
+[id="ocp-4-2-OVN"]
+==== OVN (Technology Preview)
+
+Open Virtual Networking (OVN) for Open vSwitch, currently in
+xref:ocp-4-2-technology-preview[Technology Preview], has many advantages,
+including acceleration of customer-driven feature requirements, some of which
+are pre-enabled, including:
+
+* Low barrier to integration; an implementation of virtual networking via OVS.
+* SDN portfolio consolidation.
+* Virtually eliminate iptables scale issues.
+* Heterogeneous Linux/Windows clusters.
+* The ability to have a cluster network that spans on-premise nodes and cloud
+nodes.
+* Full Network Policy support.
+* Egress IP per pod.
+* Distributed L4 Ingress/Egress firewall.
+* Distributed services load balancer.
+* Heterogeneous clusters with Windows nodes
+* The capability to span on-premise and cloud nodes.
+* Traffic isolation and multi-tenancy.
+* Data Plane Development Kit (DPDK) support.
+* Encrypted tunnels.
+* IPv6 and DHCPv6.
+* QoS, control and data plane separation.
+
+[id="ocp-4-2-enable-ingress-controllers"]
+==== Enable internal ingress controllers for private clusters
```

Comment:
Suggested replacement for this section:

When creating an IngressController on cloud platforms, the IngressController is
published by a public cloud load balancer by default.

Users can now publish IngressControllers with internal cloud load balancers. For example:

```
apiVersion: operator.openshift.io/v1
kind: IngressController
metadata:
  namespace: openshift-ingress-operator
  name: internal
spec:
  endpointPublishingStrategy:
    type: LoadBalancerService
    loadBalancer:
      scope: Internal
```

See the link:https://kubernetes.io/docs/concepts/services-networking/#internal-load-balancer.[Kubernetes Services documentation]
for implementation details.

Note that once set, `.spec.endpointPublishingStrategy.loadBalancer.scope` cannot
be changed. To change the scope, delete and recreate the IngressController.

The `default` IngressController can be made internal by deleting and recreating
it.

Reply:
cc @openshift/sig-network-edge

Reply:
Since this is release notes, I think we should move the details and example to product document (a new topic in https://docs.openshift.com/container-platform/4.2/networking/ingress-operator.html), and here just provide the link to it like: 
`See xref:.. for more information.`

Reply:
So I've created https://github.com/openshift/openshift-docs/pull/16963 for this content.